instrumentor/controllers/agentenabled/pods_webhook.go

```diff
@@ -358,3 +386,50 @@ func getRuntimeInfoForContainerName(ic *odigosv1.InstrumentationConfig, containe
 	// if both are not found, return we don't have runtime info for this container
 	return nil
 }
+
+func CreateInitContainer(pod *corev1.Pod, dirsToCopy map[string]struct{}, config common.OdigosConfiguration) {
+	const (
+		instrumentationsPath = "/instrumentations"
+	)
+	imageVersion := os.Getenv(consts.OdigosVersionEnvVarName)
+	imageName := config.ImagePrefix + "/" + k8sconsts.OdigosInitContainerName + ":" + imageVersion
+
+	if len(dirsToCopy) == 0 {
+		return
+	}
```

Comment:
should we check this before calling the function? so we also know to skip other related mutations